plat/nvidia/tegra/platform.mk

```diff
@@ -1,9 +1,19 @@
 #
-# Copyright (c) 2015-2017, ARM Limited and Contributors. All rights reserved.
+# Copyright (c) 2015-2018, ARM Limited and Contributors. All rights reserved.
 #
 # SPDX-License-Identifier: BSD-3-Clause
 #
 
```

Comment:
I still don't know why this patch is needed. All libc files are compiled and packaged in a library archive. As far as I understand, linkers only pull in the object files that are actually used. This isn't even about removing individual functions in one object file, this is about removing the object file itself.

Also, this patch relies on the current order of inclusions in the Makefile. If the standard ``LIBC_SRCS`` is expanded before this file is included, it will have no effect.

Reply:
I feel that this may result in a conversation similar to the one in https://github.com/ARM-software/arm-trusted-firmware/pull/1751. If you still think it is needed, maybe it's worth moving it to that PR. I wouldn't want to block this PR just because of a discussion in one patch that doesn't affect the others.

Reply:
@antonio-nino-diaz-arm 

This is required to get to zero MISRA defects for Tegra. Since we do not use these files, it reduces the amount of MISRA defects we have to deal with. Plus, this patch also reduces the code size.

> Also, this patch relies on the current order of inclusions in the Makefile. If the standard LIBC_SRCS is expanded before this file is included, it will have no effect.

That's true. We can either document that dependency or change this makefile to remove the files from BL31_SOURCES

Reply:
Hi Varun,
Not sure if I understand the code size argument. The libc is built as a `library archive` and only referenced symbols will be included in the binary. Also the `--gc-sections` will take care of removing unused sections.

The MISRA defect argument is justified but you seem to have included most of files except some string utilities and I don't think there would be many MISRA defects in the other files excluded. Fixing the few remaining MISRA defects in the other files shouldn't be too difficult.

Allowing platforms to override makefile variables like these and adding dependency on the order of inclusion for building library archives will not allow us to make improvements/changes to the build system without affecting platform ports. Hence currently I don't see a convincing reason why this should be allowed.